app/ante/fee_checker.go

```diff
@@ -23,27 +23,34 @@ func CheckTxFeeWithGlobalMinGasPrices(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, in
 		return nil, 0, errors.Wrap(sdkerror.ErrTxDecode, "Tx must be a FeeTx")
 	}
 
-	fee := feeTx.GetFee().AmountOf(appconsts.BondDenom)
-	gas := feeTx.GetGas()
-
 	// global minimum fee only applies to app versions greater than one
 	if ctx.BlockHeader().Version.App > v1.Version {
-		// convert the global minimum gas price to a big.Int
-		globalMinGasPrice, err := sdk.NewDecFromStr(fmt.Sprintf("%f", appconsts.GlobalMinGasPrice))
+		err := CheckTxFeeWithMinGasPrices(feeTx, appconsts.GlobalMinGasPrice, "insufficient global minimum fee")
```

Comment:
[blocking question]

is this called during processproposal? if so we need to not to avoid any non-determinism. I didn't see a test for this but could have missed it, but that's something that we probably need. ProcessProposal should have the same result for a block with txs that are below the locally configured value

Reply:
`CheckTxFeeWithGlobalMinGasPrices` is called in processproposal through an antehandler, checking the global fee